extensions/amp-social-share/0.2/social-share.js

```diff
@@ -91,3 +134,64 @@ export function SocialShare(props) {
     </div>
   );
 }
+
+/**
+ * @param {!JsonObject} props
+ * @param {?string} name
+ * @return {!JsonObject}
+ */
+function checkProps(props, name) {
+  const {type, shareEndpoint, params, bindings, width, height} = props;
+
+  // Verify type is provided
+  if (type === undefined) {
+    throw new Error(`The type attribute is required. ${name}`);
+  }
+
+  // bindings and params props must be objects
+  if (params && !isObject(params)) {
+    throw new Error(`The params property should be an object. ${name}`);
+  }
+  if (bindings && !isObject(bindings)) {
+    throw new Error(`The bindings property should be an object. ${name}`);
+  }
+
+  // User must provide shareEndpoint if they choose a type that is not
+  // pre-configured
+  const typeConfig = getSocialConfig(type) || dict();
+  const baseEndpoint = typeConfig['shareEndpoint'] || shareEndpoint;
+  if (baseEndpoint === undefined) {
+    throw new Error(
+      `A shareEndpoint is required if not using a pre-configured type. ${name}`
+    );
+  }
+
+  // Verify width and height are valid integers
+  let checkedWidth = Math.floor(width || DEFAULT_WIDTH);
```

Comment:
This is an interesting case: `Math.floor(null)` yields `0` but `Math.floor(undefined)` yields `NaN` -- if the prop isn't present it's `undefined`, meaning it would be reasonable to rewrite this: `Math.floor(width) || DEFAULT_WIDTH`

Do we ever expect `width` to be `null`? That would be a reasonable case for the current order.

Reply:
Edit: Just saw the positive integer check below, in which case even if it is `null` we would prefer to use the `DEFAULT_WIDTH`.

tldr supernit: `Math.floor(width) || DEFAULT_WIDTH`

Reply:
Ahh thanks for the input.  Right, I agree there are many tricky corner cases for this.  I've included your input in the new version which now follows the below behavior!

  // Verify width and height are valid integers
  // Silently assigns default for undefined, null
  // Throws Warning for booleans, strings, 0, negative numbers
  // No errors when positive integer or equivalent string